includes/wikia/tests/WikiaTest.php

```diff
@@ -8,6 +11,48 @@ public function setUp() {
 		parent::setUp();
 	}
 
+	/**
+	 * @param LocalFile $fileMock
+	 * @param string $expectedUrl
+	 * @param string $expectedSize
+	 *
+	 * @dataProvider getWikiLogoMetadataDataProvider
+	 */
+	public function testGetWikiLogoMetadata( $fileMockData, $expectedUrl, $expectedSize ) {
+		$fileMock = $this->mockClassWithMethods( 'stdClass', $fileMockData );
+
+		$this->mockGlobalFunction( 'wfLocalFile', $fileMock );
+		$this->mockGlobalVariable( 'wgResourceBasePath', 'http://wikia.net/__cb123' );
+
+		$logoData = Wikia::getWikiLogoMetadata();
+
+		$this->assertEquals( $expectedUrl, $logoData['url'] );
+		$this->assertEquals( $expectedSize, $logoData['size'] );
+	}
+
+	public function getWikiLogoMetadataDataProvider() {
+		return [
+			[
+				'fileMockData' => [
+					'exists' => false,
+				],
+				// see wgResourceBasePath mock above
+				'expectedUrl' => 'http://wikia.net/__cb123/skins/common/images/wiki.png',
+				'expectedSize' => '155x155',
+			],
+			[
+				'fileMockData' => [
+					'exists' => true,
+					'getUrl' => '/foo.png',
+					'getWidth' => 42,
+					'getHeight' => 32,
+				],
+				'expectedUrl' => '/foo.png',
+				'expectedSize' => '42x32',
+			],
+		];
+	}
+
 	public function testOnSkinTemplateOutputPageBeforeExec_setsNoIndexNoFollowBecauseHeaders() {
 		$this->mockGlobalVariable('wgDevelEnvironment', false);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "wgDevelEnvironment" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)